cuqi/experimental/mcmc/_conjugate.py

```diff
@@ -165,9 +172,107 @@ def sample(self):
         beta = self.target.prior.rate                                   # beta
 
         # Create Gamma distribution and sample
-        dist = Gamma(shape=m/2 + alpha, rate=.5 * np.linalg.norm(L @ (Ax - b))**2 + beta)
+        return Gamma(shape=m/2 + alpha, rate=.5 * np.linalg.norm(L @ (Ax - b))**2 + beta)
+    
+    
+class _RegularizedUnboundedUniformGammaPair(_ConjugatePair):
+    """Implementation for the RegularizedUnboundedUniform-ModifiedHalfNormal conjugate pair."""
 
-        return dist.sample()
+    def validate_target(self):
+        if self.target.prior.dim != 1:
+            raise ValueError("RegularizedUnboundedUniform-Gamma conjugacy only works with univariate Gamma prior")
+        
+        if self.target.likelihood.distribution.preset not in ["l1", "tv"]:
+            raise ValueError("RegularizedUnboundedUniform-Gamma conjugacy only works with implicit regularized Gaussian likelihood with l1 or tv regularization")
+        
+        key_value_pairs = _get_conjugate_parameter(self.target)
+        if len(key_value_pairs) != 1:
+            raise ValueError(f"Multiple references to conjugate parameter {self.target.prior.name} found in likelihood. Only one occurance is supported.")
+        for key, value in key_value_pairs:
+            if key == "strength":
+                if not _check_conjugate_parameter_is_scalar_linear(value):
+                    raise ValueError("RegularizedUnboundedUniform-Gamma conjugacy defined via strength requires strength: lambda x : s*x for the conjugate parameter")
+            else:
+                raise ValueError(f"RegularizedUnboundedUniform-Gamma conjugacy does not support the conjugate parameter {self.target.prior.name} in the {key} attribute. Only strength is supported")
+
+    def conjugate_distribution(self):
+        # Extract prior variables
+        alpha = self.target.prior.shape
+        beta = self.target.prior.rate
+
+        # Compute likelihood quantities
+        x = self.target.likelihood.data
+        if self.target.likelihood.distribution.preset == "l1":
+            m = count_nonzero(x)
+        elif self.target.likelihood.distribution.preset == "tv" and isinstance(self.target.likelihood.distribution.geometry, Continuous1D):
+            m = count_constant_components_1D(x)
+        elif self.target.likelihood.distribution.preset == "tv" and isinstance(self.target.likelihood.distribution.geometry, (Continuous2D, Image2D)):
+            m = count_constant_components_2D(self.target.likelihood.distribution.geometry.par2fun(x))
+
+        reg_op = self.target.likelihood.distribution._regularization_oper
+        reg_strength = self.target.likelihood.distribution(np.array([1])).strength
+        fx = reg_strength*np.linalg.norm(reg_op@x, ord = 1)
+
+        # Create Gamma distribution and sample
```

Comment:
We only set the distribution, no sample here, right?
```suggestion
        # Create Gamma distribution
```